### PR TITLE
Refactor some `Http1Connection` methods

### PR DESCRIPTION
**Pull request by danicheg**
This refactors some methods of the `Http1Connection` to simplify reasoning. Implementation, by and large, wasn't changed. Just decomposition was made more fine-grained and nested ifs replaced by pattern-matching.

### CODE REVIEW

File: blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala [hunk 1]
```diff
@@ -310,105 +310,113 @@ private final class Http1Connection[F[_]](
         cb(Left(t))
     }(executionContext)
 
-  private def parsePrelude(
+  // it's called when the body is known
+  private def terminationConditionParsingPrelude(): Either[Throwable, Option[Chunk[Byte]]] =
```
Comment (RafalSumislawski) [inline blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala:314]: I would like to search for a better name for this method. It actually is relevant to parsing the body not the prelude. It's called when an EOF is received while parsing body, and the purpose of this metod it to decide whether it's ok that we recieved the EOF or not.

Maybe something like `onEofWhileReadingBody`?

File: blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala [hunk 2]
```diff
@@ -310,105 +310,113 @@ private final class Http1Connection[F[_]](
...
+    stageState.get match { // if we don't have a length, EOF signals the end of the body.
+      case Error(e) if e != EOF => Either.left(e)
+      case _ =>
+        if (parser.definedContentLength() || parser.isChunked())
+          Either.left(InvalidBodyException("Received premature EOF."))
+        else Either.right(None)
+    }
+
+  private def cleanupParsingPrelude(closeOnFinish: Boolean, headers: Headers): Unit =
```
Comment (RafalSumislawski) [inline blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala:323]: Maybe `cleanupParsingResponse`? Similarly to my comment on `terminationConditionParsingPrelude`. The `cleanupParsingPrelude` happens later than after parsing prelude. First we parse the prelude, then the body, then (optionally) the trailers and then we clean up.

IMO it should be `cleanUp` (a verb) not `cleanup` (a noun).

File: blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala [hunk 3]
```diff
@@ -310,105 +310,113 @@ private final class Http1Connection[F[_]](
...
+          headers = headers,
+          body = body.interruptWhen(idleTimeoutS),
+          attributes = attributes,
         )
       )
-    } catch {
+    )
+  }
+
+  private def parsePrelude(
```
Comment (RafalSumislawski) [inline blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala:406]: IMO this class would be easier if reading it top to bottom would be consistent with the order in which these operations happen (methods are called). The already existing methods are ordered like this. At least the most important ones.

I would like to see them in this order:
- runRequest
- executeRequest
- receiveResponse
- handleRead
- parsePrelude
- parsePreludeFinished (new)
- terminationConditionParsingPrelude (new)
- cleanupParsingPrelude (new)

**Review by RafalSumislawski — CHANGES_REQUESTED** [review]

**Comment by danicheg** [comment]
@RafalSumislawski wow, what an utter review! :) If I get it right, you argue on most about naming, formatting, and order of methods. I give or take ok with your concerns, so feel free to proceed with my work on this if you want it. I'm not so interested in pushing this further.

File: blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala [hunk 4]
```diff
@@ -310,105 +310,113 @@ private final class Http1Connection[F[_]](
...
+      buffer: ByteBuffer,
+      closeOnFinish: Boolean,
+      doesntHaveBody: Boolean,
+      cb: Callback[Response[F]],
+      idleTimeoutS: F[Either[Throwable, Unit]],
+  ): Unit =
+    try if (!parser.finishedResponseLine(buffer))
```
[the remark this answers is not in the document]
Comment (danicheg) [reply blaze-client/src/main/scala/org/http4s/blaze/client/Http1Connection.scala:413]: Probably this should be done by Scalafmt? Though it's not looking so bizarre for me.

**Review by RafalSumislawski — APPROVED** [review]

**Comment by RafalSumislawski** [comment]
Hi @danicheg 
I applied the changes that I suggested earlier in the review. Please check if you're ok with the changes before we merge.

**Comment by danicheg** [comment]
@RafalSumislawski thanks. LGTM.

**Pull request merged**